catalyst/dl/experiment/gan.py

```diff
@@ -0,0 +1,122 @@
+from typing import Iterable, Dict, Mapping, Any, Union, OrderedDict
+from torch.utils.data import DataLoader
+from catalyst.dl.core import Experiment
+from catalyst.dl.callbacks import (
+    CheckpointCallback,
+    ConsoleLogger,
+    RaiseExceptionCallback,
+    VerboseLogger,
+)
+from catalyst.dl.utils import process_callbacks
+from catalyst.utils.typing import (
+    Model,
+    Criterion,
+    Optimizer,
+    Scheduler,
+    Dataset
+)
+
+
+class GanExperiment(Experiment):
+    def __init__(
+        self,
+        models: Union[Model, Dict[str, Model]],
```

Comment:
in 
```
def get_model(self, stage: str) -> Model:
        return self._models[stage]
```
you suppose that `models` can be only a dict

Reply:
In typing.py I've set Model = Union[torch.nn.Module, Dict[str, torch.nn.Module]].
Yeah, good point to change that one too.